py/server/deephaven/partitioned_table.py

```diff
@@ -0,0 +1,1175 @@
+            match_rule (AsOfMatchRule): the inexact matching rule on the last column to match specified in 'on',
+                default is AsOfMatchRule.LESS_THAN_EQUAL. The other valid value is AsOfMatchRule.LESS_THAN.
+        Returns:
+            a new PartitionedTableProxy
+
+        Raises:
+            DHError
+        """
+        try:
+            on = to_sequence(on)
+            joins = to_sequence(joins)
+            if on:
+                on = [_JJoinMatch.parse(p) for p in on]
+            if joins:
+                joins = [_JJoinAddition.parse(p) for p in joins]
+
+            on = j_array_list(on)
+            joins = j_array_list(joins)
+
+            if match_rule is AsOfMatchRule.LESS_THAN_EQUAL:
+                match_rule = _JAsOfJoinRule.LESS_THAN_EQUAL
+            else:
+                match_rule = _JAsOfJoinRule.LESS_THAN
+
+            return PartitionedTableProxy(j_pt_proxy=self.j_pt_proxy.aj(table.j_table, on, joins, match_rule))
+        except Exception as e:
+            raise DHError(e, "as-of join operation on the PartitionedTableProxy failed.") from e
+
+    @auto_locking_op
+    def raj(self, table: Table, on: Union[str, Sequence[str]], joins: Union[str, Sequence[str]] = None,
+            match_rule: AsOfMatchRule = AsOfMatchRule.GREATER_THAN_EQUAL) -> PartitionedTableProxy:
+        """Applies the table 'raj' operation to all constituent tables of the underlying partitioned
+        table with the provided right table, and produces a new PartitionedTableProxy with the result tables as the
+        constituents of its underlying partitioned table.
+
+        Args:
+            table (Table): the right-table of the join
+            on (Union[str, Sequence[str]]): the column(s) to match, can be a common name or an equal expression,
+                i.e. "col_a = col_b" for different column names
+            joins (Union[str, Sequence[str]], optional): the column(s) to be added from the right table to the result
+                table, can be renaming expressions, i.e. "new_col = col"; default is None
+            match_rule (AsOfMatchRule): the inexact matching rule on the last column to match specified in 'on',
+                default is AsOfMatchRule.GREATER_THAN_EQUAL. The other valid value is AsOfMatchRule.GREATER_THAN.
+        Returns:
+            a new PartitionedTableProxy
+
+        Raises:
+            DHError
+        """
+        try:
+            on = to_sequence(on)
+            joins = to_sequence(joins)
+            if on:
+                on = [_JJoinMatch.parse(p) for p in on]
+            if joins:
+                joins = [_JJoinAddition.parse(p) for p in joins]
+
+            on = j_array_list(on)
+            joins = j_array_list(joins)
+
+            if match_rule is AsOfMatchRule.GREATER_THAN_EQUAL:
+                match_rule = _JReverseAsOfJoinRule.GREATER_THAN_EQUAL
+            else:
+                match_rule = _JReverseAsOfJoinRule.GREATER_THAN
```

Comment:
I would make this an explicit check and then raise an error for the default case to protect us in case someone changes the java.